chore(api): disable legacy calibration accessors on flex

sfoster1 opened the pull request:
We had a lot of spammy logs in the api log about missing or malformed pipette and tip length calibration.

The missing calibrations should not be logged at info level - these are polled and this is a level state, so it just ends up being spammy. The malformed calibrations though should be, and it's weird that we were getting them.

The culprit is that <7.0.0 apps poll routes to get pipette offset and tip length because that's what you do on the ot2, and the implementations of those routes hardcode the ot2 calibration models - reasonable, since we completely replaced them for the flex. So you'd only get those logs if an old app was polling the robot.

To fix that, I made the routes shortcircuit with a 403 forbidden / not allowed on hardware error. This won't affect the flex because the app doesn't check that route on flex.

sfoster1 committed:
chore(api): remove spammy log messages

These messages about missing calibrations are completely meaningless on
the flex and are incredibly spammy; pull them.

They might have meaning on the OT-2 but honestly we have a
_lot_ of ui surrounding them so whatever.

SyntaxColoring commented on api/src/opentrons/calibration_storage/ot2/pipette_offset.py:
```diff
@@ -88,12 +88,13 @@ def get_pipette_offset(
         return None
     except (json.JSONDecodeError, ValidationError):
-        log.warning(
+        log.debug(
```
I might leave this as a warning, unless this is actually more false-positive than true-positive in practice.

sfoster1 replied:
this is, unfortunately, a frequent occurrence when a 6.3.1 app is viewing a flex because it's polling the old pipette offset route. which i might want to just disable in the server or something

SyntaxColoring commented on api/src/opentrons/calibration_storage/ot3/tip_length.py:
```diff
@@ -37,19 +37,22 @@ def tip_lengths_for_pipette(
         all_tip_lengths_for_pipette = io.read_cal_file(tip_length_filepath)
         for tiprack, data in all_tip_lengths_for_pipette.items():
             try:
                 tip_lengths[tiprack] = v1.TipLengthModel(**data)
             except (json.JSONDecodeError, ValidationError):
-                log.warning(
+                log.debug(
                     f"Tip length calibration is malformed for {tiprack} on {pipette_id}"
```
I might leave this as a warning, unless this is actually more false-positive than true-positive in practice.

sfoster1 replied:
this just never should ever print ever so i can make it a warning

SyntaxColoring reviewed (approved):
🥫

sfoster1 committed:
more debug logs

sfoster1 committed:
Refuse to get offset and tl on flex

Make the legacy routes for tip_length and pipette_offset 503 on flex and
not try and get the offsets using the old calibration accessors, since
they are hardcoded to use the ot2 versions of calibration which do not
apply to the flex and dumped errors in the logs all the time.

the api side change supports testing.

SyntaxColoring commented on robot-server/robot_server/service/pipette_offset/router.py:
```diff
@@ -45,7 +48,9 @@ def _format_calibration(
     response_model=pip_models.MultipleCalibrationsResponse,
 )
 async def get_all_pipette_offset_calibrations(
-    pipette_id: Optional[str] = None, mount: Optional[pip_models.MountType] = None
+    pipette_id: Optional[str] = None,
+    mount: Optional[pip_models.MountType] = None,
+    _: API = Depends(get_ot2_hardware),
```
To make sure I understand correctly: on a Flex, even *retrieving* pipette calibrations through these endpoints was doing the wrong thing?

sfoster1 replied:
Yes. This code was hardcoded to select the ot2 pipette offset loaders.

SyntaxColoring reviewed (approved):
Agreed with blocking off these routes if they aren't how calibration works on the Flex. Thanks!

sfoster1 committed:
forgot to save this file
